geoportailv3/static/js/draw/drawnfeaturesservice.js

```diff
@@ -133,6 +132,26 @@ app.DrawnFeatures.prototype.saveFeature = function(feature) {
 
 
 /**
+ * Copy anonymous features to mymaps
+ * @return {angular.$q.Promise} Promise.
+ */
+app.DrawnFeatures.prototype.copyAnonymousFeaturesToMymaps = function() {
+  var anonymousFeatures = this.features.getArray().
+      filter(function(feature) {
+        return feature.get('__source__') !== 'mymaps';
+      });
+  anonymousFeatures.forEach(goog.bind(function(feature) {
+    feature.set('__source__', 'mymaps');
+  },this));
+
+  return this.saveFeaturesInMymaps_(anonymousFeatures).then(
+      goog.bind(function(resp) {
+        return resp;
+      }, this));
```

Comment:
And why returning the promise returned by `then` instead of directly returning the promise returned by `saveFeaturesInMymaps_`?